src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -2020,11 +2020,28 @@ internal Task<Compilation> GetCompilationAsync(Project project, CancellationToke
                 : SpecializedTasks.Default<Compilation>();
         }
 
-        internal Task<bool> HasCompleteReferencesAsync(Project project, CancellationToken cancellationToken)
+        /// <summary>
+        /// Return reference completeness for the given project.
+        /// </summary>
+        internal Task<bool> IsCompleteAsync(Project project, CancellationToken cancellationToken)
+        {
+            // return IsComplete when compilation is not supported. 
+            // regardless whether project support compilation or not, if projectInfo is not complete, we can't gurantee its reference completeness
+            return project.SupportsCompilation
+                ? this.GetCompilationTracker(project.Id).IsCompleteAsync(this, cancellationToken)
+                : project.IsComplete ? SpecializedTasks.True : SpecializedTasks.False;
```

Comment:
we dont want to put reference in API since intellisense build failure is not tied to references. we do care about references due to skeleton assembly failure but, that doesnt make API need to specifically tells about references.

HasSuccessfullyLoaded and HasSuccessfullyLoadedTransitively seems good.